sync2/handler2/handler.go

```diff
@@ -274,6 +274,7 @@ func (h *Handler) Accumulate(ctx context.Context, userID, deviceID, roomID strin
 	for i := range timeline.Events {
 		// Delete MSC4115 field as it isn't accurate when we reuse the same event for >1 user
 		timeline.Events[i], _ = sjson.DeleteBytes(timeline.Events[i], "unsigned.membership")
+		timeline.Events[i], _ = sjson.DeleteBytes(timeline.Events[i], `unsigned.io\.element\.msc4115\.membership`)
```

Comment:
Why do not all `.` need to be `\.`? (the one between `unsigned.io` isn't)
Is this specific to how synapse formats things?
Might be worth a comment explaining where these `\` come from.

Reply:
Because `.` means "look at a field in this object" which makes it impossible to refer to keys which actually have "."s in them. To do this, you need to escape them e.g `foo.bar\.baz` refers to the value "quuz":
```json
{
  "foo": {
    "bar.baz": "quuz"
  }
}
```

Reply:
ah I see. makes sense